pkgs/by-name/go/golden-cheetah-bin/package.nix

```diff
@@ -1,38 +1,92 @@
-    hash = "sha256-PMRUDQSQxbECbF9SPOo03t4Xxj1OtYJAPXEMyyy6EVY=";
+    url = url_for_platform version;
+    hash =
+      if stdenv.hostPlatform.isDarwin then
+        "sha256-GGe8pGf16FWqN9uaStnGOtOeKDB44VBD7QwHkaZb7wY="
+      else
+        "sha256-S1YDg7opoYflL/3h+lP1SNwXpJHaN3+iE3waSjWtN6o=";
   };
-
-  appimageContents = appimageTools.extract { inherit pname src version; };
-in
-appimageTools.wrapType2 {
-  inherit pname src version;
-
-  extraPkgs = pkgs: [ pkgs.R pkgs.zlib pkgs.libusb-compat-0_1 ];
-
-  extraInstallCommands = ''
-    mv $out/bin/${pname} $out/bin/GoldenCheetah
-    mkdir -p $out/share/applications
-    mkdir -p $out/share/pixmaps
-    cp ${appimageContents}/GoldenCheetah.desktop $out/share/applications/
-    cp ${appimageContents}/gc.png $out/share/pixmaps/
-  '';
-
-  passthru.updateScript = nix-update-script { };
-
   meta = {
     description = "Performance software for cyclists, runners and triathletes. This version includes the API Tokens for e.g. Strava";
-    platforms = lib.platforms.linux;
-    broken = !stdenv.hostPlatform.isx86_64;
-    maintainers = with lib.maintainers; [ gador adamcstephens ];
+    platforms = lib.platforms.unix;
+    broken = if !stdenv.hostPlatform.isDarwin then !stdenv.hostPlatform.isx86_64 else false;
+    maintainers = with lib.maintainers; [
+      gador
+      adamcstephens
+    ];
     license = lib.licenses.gpl2Plus;
     sourceProvenance = with lib.sourceTypes; [ binaryNativeCode ];
-    knownVulnerabilities = [ "Vendors libwebp vulnerable to CVE-2023-4863" ];
   };
-}
+in
+
+if stdenv.hostPlatform.isDarwin then
+  stdenv.mkDerivation {
+    inherit
+      pname
+      src
+      version
+      meta
+      ;
+
+    nativeBuildInputs = [
+      makeWrapper
+      undmg
+    ];
+    dontBuild = true;
+    installPhase = ''
+      runHook preInstall
+      mkdir -p "$out/Applications/GoldenCheetah.app"
+      cp -r * "$out/Applications/GoldenCheetah.app/"
+      makeWrapper "$out/Applications/GoldenCheetah.app/Contents/MacOS/GoldenCheetah" "$out/bin/GoldenCheetah"
+      runHook postInstall
+    '';
+  }
+else
+  appimageTools.wrapType2 rec {
+    inherit
+      pname
+      src
+      version
+      meta
+      ;
+
+    extraPkgs = pkgs: [
+      pkgs.R
+      pkgs.zlib
+      pkgs.libusb-compat-0_1
+    ];
+
+    appimageContents = appimageTools.extract { inherit pname src version; };
+
+    extraInstallCommands = ''
+      mv $out/bin/${pname} $out/bin/GoldenCheetah
+      mkdir -p $out/share/applications
+      mkdir -p $out/share/pixmaps
+      cp ${appimageContents}/GoldenCheetah.desktop $out/share/applications/
+      substituteInPlace $out/share/applications/GoldenCheetah.desktop --replace-fail \
+        "Exec=GoldenCheetah" "Exec=QT_PLUGIN_PATH= GoldenCheetah"
+      cp ${appimageContents}/gc.png $out/share/pixmaps/
+    '';
+
+    passthru.updateScript = nix-update-script { };
```

Comment:
```suggestion
```
I don't believe this will work in the current setup

Reply:
Yes. It actually didn't really work before either, since upstream changed the version format. I removed it